docs/guidance/kubeflow-integration.md

```diff
@@ -0,0 +1,106 @@
+> Credit: This manifest refers a lot to the engineering blog ["Building a Machine Learning Platform with Kubeflow and Ray on Google Kubernetes Engine"](https://cloud.google.com/blog/products/ai-machine-learning/build-a-ml-platform-with-kubeflow-and-ray-on-gke) from Google Cloud.
+
+# Kubeflow: an interactive development solution
+
+# Requirements
+* Dependencies
+    * `kustomize`: v3.2.0 (Kubeflow manifest is sensitive to `kustomize` version.)
+    * `Kubernetes`: v1.23
+
+* Computing resources:
+    * 16GB RAM
+    * 8 CPUs
+
+# Example: Use Kubeflow to provide an interactive development envirzonment
+![image](../images/architecture.svg)
+
+## Step 1: Create a Kubernetes cluster with Kind.
+```sh
+# Kubeflow is sensitive to Kubernetes version and Kustomize version.
+kind create cluster --image=kindest/node:v1.23.0
+kustomize version --short
+# 3.2.0
+```
+
+## Step 2: Install Kubeflow v1.6-branch
+* This example installs Kubeflow with the [v1.6-branch](https://github.com/kubeflow/manifests/tree/v1.6-branch).
+
+* Install all Kubeflow official components and all common services using [one command](https://github.com/kubeflow/manifests/tree/v1.6-branch#install-with-a-single-command).
+    * If you do not want to install all components, you can comment out **KNative**, **Katib**, **Tensorboards Controller**, **Tensorboard Web App**, **Training Operator**, and **KServe** from [example/kustomization.yaml](https://github.com/kubeflow/manifests/blob/v1.6-branch/example/kustomization.yaml).
+
+## Step 3: Install KubeRay operator
+* Follow this [document](../../helm-chart/kuberay-operator/README.md) to install the latest stable KubeRay operator via Helm repository.
+
+## Step 4: Install RayCluster
+```sh
+# Create a RayCluster CR, and the KubeRay operator will reconcile a Ray cluster
+# with 1 head Pod and 1 worker Pod.
+helm install raycluster kuberay/ray-cluster --version 0.4.0 --set image.tag=2.2.0-py38-cpu
+
+# Check RayCluster
+kubectl get pod -l ray.io/cluster=raycluster-kuberay
+# NAME                                          READY   STATUS    RESTARTS   AGE
+# raycluster-kuberay-head-bz77b                 1/1     Running   0          64s
+# raycluster-kuberay-worker-workergroup-8gr5q   1/1     Running   0          63s
+```
+* This step uses `rayproject/ray:2.2.0-py38-cpu` as its image. Ray is very sensitive to the Python versions and Ray versions between the server (RayCluster) and client (JupyterLab) sides. This image uses:
+    * Python 3.8.13
+    * Ray 2.2.0
```

Comment:
I suppose we can use 2.3.0 now, but if it will require a lot more manual testing then it might not be worth it.

Reply:
It is a huge effort. Ray client is very sensitive to the Python versions and Ray versions. Sometimes both client and server images use same commit, but it is still not compatible.